File: app/src/main/java/zebrostudio/wallr100/ui/explore/ExploreModule.java
package zebrostudio.wallr100.ui.explore;


import android.content.Context;

import dagger.Module;
import dagger.Provides;
import zebrostudio.wallr100.data.DataManager;
import zebrostudio.wallr100.di.ActivityScope;
import zebrostudio.wallr100.di.ApplicationContext;

@ActivityScope
@Module
public class ExploreModule {

    @Provides
    ExplorePresenterImpl providesExplorePresenterImpl(@ApplicationContext Context context,
                                                      DataManager dataManager) {
        return new ExplorePresenterImpl(context, dataManager);
    }
}

File: app/src/main/java/zebrostudio/wallr100/android/ui/wallpaper/explore/ExploreModule.kt
package zebrostudio.wallr100.android.ui.wallpaper.explore

import dagger.Module
import dagger.Provides
import zebrostudio.wallr100.domain.interactor.WallpaperImagesUseCase
import zebrostudio.wallr100.presentation.wallpaper.explore.ExploreContract
import zebrostudio.wallr100.presentation.wallpaper.explore.ExplorePresenterImpl
import zebrostudio.wallr100.presentation.wallpaper.mapper.ImagePresenterEntityMapper

@Module
class ExploreModule {

  @Provides
  fun provideImagePresenterEntityMapper(): ImagePresenterEntityMapper = ImagePresenterEntityMapper()

  @Provides fun provideExplorePresenter(
    wallpaperImagesUseCase: WallpaperImagesUseCase,
    imagePresenterEntityMapper: ImagePresenterEntityMapper
  ): ExploreContract.ExplorePresenter = ExplorePresenterImpl(wallpaperImagesUseCase,
      imagePresenterEntityMapper)
}